packages/cli/src/controllers/passwordReset.controller.ts

```diff
@@ -187,27 +189,52 @@ export class PasswordResetController {
 			throw new BadRequestError('');
 		}
 
-		// Timestamp is saved in seconds
-		const currentTimestamp = Math.floor(Date.now() / 1000);
+		let decodedToken: jwt.JwtPayload;
+
+		try {
+			decodedToken = jwt.verify(resetPasswordToken, config.getEnv('userManagement.jwtSecret'), {
+				ignoreExpiration: false,
+			}) as jwt.JwtPayload;
+		} catch (e) {
+			if (e instanceof TokenExpiredError) {
+				this.logger.debug('Reset password token expired', {
+					userId: id,
+					resetPasswordToken,
+				});
+				throw new NotFoundError('');
+			}
+			throw new BadRequestError('');
+		}
+
+		if (id !== decodedToken.sub) {
```

Comment:
since the token already contains the user's id, we should remove `userId` from the querystring, and also remove this check. If a jwt signature is valid, we should just trust the userId from the decoded token.

Reply:
good catch!